compute/snapshots_test.go

```diff
@@ -0,0 +1,135 @@
+	_SnapshotInstanceTestImage = "/oracle/public/JEOS_OL_6.6_10GB_RD-1.2.217-20151201-194209"
+)
+
+func TestAccSnapshotLifeCycleBasic(t *testing.T) {
+	helper.Test(t, helper.TestCase{})
+
+	sClient, iClient, mClient, err := getSnapshotsTestClients()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// In order to get details on a Snapshot we need to create the following resources
+	// - Instance
+
+	instanceInput := &CreateInstanceInput{
+		Name:      _SnapshotInstanceTestName,
+		Label:     _SnapshotInstanceTestLabel,
+		Shape:     _SnapshotInstanceTestShape,
+		ImageList: _SnapshotInstanceTestImage,
+	}
+
+	createdInstance, err := iClient.CreateInstance(instanceInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDownInstances(t, iClient, createdInstance.Name, createdInstance.ID)
+
+	createSnapshotInput := &CreateSnapshotInput{
+		Instance: strings.Join([]string{createdInstance.Name, createdInstance.ID}, "/"),
+	}
+	createdSnapshot, err := sClient.CreateSnapshot(createSnapshotInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDownSnapshots(t, sClient, mClient, createdSnapshot.Name, createdSnapshot.MachineImage)
+
+	getInput := &GetSnapshotInput{
+		Name: createdSnapshot.Name,
+	}
+
+	snapshot, err := sClient.GetSnapshot(getInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("Snapshot Retrieved: %+v", snapshot)
+	if !reflect.DeepEqual(snapshot.Name, createdSnapshot.Name) {
+		t.Fatal("Snapshot Name mismatch! Got: %s Expected: %s", snapshot.Name, createdSnapshot.Name)
+	}
+}
+
+func TestAccSnapshotLifeCycleMachineImage(t *testing.T) {
+	helper.Test(t, helper.TestCase{})
+
+	sClient, iClient, mClient, err := getSnapshotsTestClients()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// In order to get details on a Snapshot we need to create the following resources
+	// - Instance
+
+	instanceInput := &CreateInstanceInput{
+		Name:      _SnapshotInstanceTestName,
+		Label:     _SnapshotInstanceTestLabel,
+		Shape:     _SnapshotInstanceTestShape,
+		ImageList: _SnapshotInstanceTestImage,
+	}
+
+	createdInstance, err := iClient.CreateInstance(instanceInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDownInstances(t, iClient, createdInstance.Name, createdInstance.ID)
+
+	createSnapshotInput := &CreateSnapshotInput{
+		Instance:     strings.Join([]string{createdInstance.Name, createdInstance.ID}, "/"),
+		MachineImage: _SnapshotInstanceTestName,
+	}
+	createdSnapshot, err := sClient.CreateSnapshot(createSnapshotInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDownSnapshots(t, sClient, mClient, createdSnapshot.Name, createdSnapshot.MachineImage)
+
+	getInput := &GetSnapshotInput{
+		Name: createdSnapshot.Name,
+	}
+
+	snapshot, err := sClient.GetSnapshot(getInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("Snapshot Retrieved: %+v", snapshot)
+	if !reflect.DeepEqual(snapshot.Name, createdSnapshot.Name) {
+		t.Fatal("Snapshot Name mismatch! Got: %s Expected: %s", snapshot.Name, createdSnapshot.Name)
+	}
+}
+
```

Comment:
We can try! I remember trying account but we might need some additional access. I'll give it a shot again.